src/main/java/htsjdk/samtools/SamFiles.java

```diff
@@ -19,61 +21,73 @@
      * @return The index for the provided SAM, or null if one was not found.
      */
     public static File findIndex(final File samFile) {
-        final File indexFile = lookForIndex(samFile); //try to find the index
-        if (indexFile == null) {
-            return unsymlinkAndLookForIndex(samFile);
+        Path path = findIndex(samFile.toPath());
+        return path == null ? null : path.toFile();
+    }
+
+    /**
+     * Finds the index file associated with the provided SAM file.  The index file must exist and be reachable to be found.
+     *
+     * If the file is a symlink and the index cannot be found, try to unsymlink the file and look for the bai in the actual file path.
+     *
+     * @return The index for the provided SAM, or null if one was not found.
+     */
+    public static Path findIndex(final Path samPath) {
```

Comment:
`findIndex(Path)` needs direct unit testing -- one test case with a regular bai index file represented as a `Path`, and another with a symlink to a bai, so that we can confirm the existing functionality has not been broken by switching to `Path` here.